doc/whatsnew/fragments/4741.feature

```diff
@@ -0,0 +1,12 @@
+A new ``json2`` reporter has been added. It features a more enriched output that is
+easier to parse and provides more info.
+
+Compared to ``json`` the only changes are that messages are now under the ``"messages"``
+key and that ``"message-id"`` now follows the camelCase convention and is renamed to
+``"messageId"``.
+The new reporter also reports the "score" of the modules you linted as defined by the
+``evaluation`` option and provides statistics about the modules you linted.
```

Comment:
I think we should just dump a json example. Descriptions of data structures are hard to follow at least for me. The minitmal example is this right ?
```json
{
    "messages": [
        {
            "type": "convention",
            "symbol": "line-too-long",
            "message": "Line too long (1/2)",
            "messageId": "C0301",
            "confidence": "HIGH",
            "module": "0123",
            "obj": "",
            "line": 1,
            "column": 0,
            "endLine": 1,
            "endColumn": 4,
            "path": "0123",
            "absolutePath": "0123"
        }
    ],
    "statistics": {
        "messageTypeCount": {
            "fatal": 0,
            "error": 0,
            "warning": 0,
            "refactor": 0,
            "convention": 1,
            "info": 0
        },
        "modulesLinted": 1,
        "score": 5.0
    }
}

```

Reply:
I can't get this to pass the `check-news fragment` check. I'd say let's skip it for now, the time investment necessary to fix it seems to be quite high for little gain.

Reply:
Fair. You could also edit the 3.0.md directly and skip making a fragment.

Reply:
Smart! Let's do that!

Reply:
Can we do that already @Pierre-Sassoulas or should we wait until the final release notes are written?

Reply:
I think we can start writing a draft of the release notes, it's going to be easier if we do it progressively.